src/InputGroup/InputGroup.js

```diff
@@ -1,272 +1,56 @@
-import Button from '../Button/Button';
 import classnames from 'classnames';
-import CustomPropTypes from '../utils/CustomPropTypes/CustomPropTypes';
-import FormInput from '../Forms/FormInput';
-import Icon from '../Icon/Icon';
+import InputGroupAddon from './_InputGroupAddon';
 import PropTypes from 'prop-types';
 import withStyles from '../utils/WithStyles/WithStyles';
-import { INPUT_GROUP_ADDON_POSITIONS, INPUT_GROUP_TYPES } from '../utils/constants';
 import React, { Component } from 'react';
 
 class InputGroup extends Component {
     constructor(props) {
         super(props);
-
-        this.state = {
-            value: this.props.inputValue,
-            searchValue: this.props.inputValue
-        };
     }
 
-    handleUp = e => {
-        e.preventDefault();
-        this.setState({
-            value: parseInt(this.state.value, 10) + 1
-        });
-    };
-
-    handleDown = e => {
-        e.preventDefault();
-        this.setState({
-            value: parseInt(this.state.value, 10) - 1
-        });
-    };
-
-    handleClear = e => {
-        e.preventDefault();
-        this.setState({
-            searchValue: ''
-        });
-    };
-
-    handleChange = e => {
-        e.preventDefault();
-        this.setState({
-            searchValue: e.target.value
-        });
-    };
-
-    handleTextChange = e => {
-        e.preventDefault();
-        this.setState({
-            value: e.target.value
-        });
-    };
-
     render() {
         const {
-            actions,
-            addonClassNames,
-            addonPos,
-            addon,
             children,
             className,
             compact,
+            customStyles,
             disableStyles,
-            glyph,
-            inputClassName,
-            inputType,
-            inputId,
-            inputName,
-            inputPlaceholder,
-            inputProps,
-            inputValue,
-            localizedText,
-            numberDownButtonProps,
-            numberUpButtonProps,
             ...props
         } = this.props;
 
+        const inputClasses = 'fd-input-group__input';
+
         const inputGroupClasses = classnames(
             className,
             'fd-input-group'
         );
 
-        const addonClasses = classnames(
-            addonClassNames,
-            'fd-input-group__addon',
-            [{ 'fd-input-group__addon--button': !!actions || inputType === 'number' }]
-        );
-
-        const inputClasses = classnames(
-            inputClassName,
-            'fd-input-group__input'
+        return (
+            <div
+                {...props}
+                className={inputGroupClasses}>
+                {React.Children.map(children, (child) => (
```

Comment:
To be safe, we should use the `toArray` syntax (like is being used in `InputGroupAddon`).
```jsx
{React.Children.toArray(children).map(child => {
```

Reply:
Sounds good - what the extra benefit is with using `React.Children.toArray(children).map` vs. `React.children.map` in this case?

Reply:
@markpalfreeman There was actually a case that someone found in `TabGroup` where the `React.Children.map` _didn't_ protect against `null` or `undefined` children, which is where this `toArray` pattern was introduced.  Whether it's actually _needed_ here could be up for debate, but I think this is a safe pattern.